extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,37 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "buffer"
+    description: >
+      Compute and return an expanded version of the input geometry. All the points
+      of the returned geometry are at a distance of `buffer_radius` away from the points
+      of the input geometry. If a negative `buffer_radius` is provided, the geometry will
+      shrink instead of expand.  A negative `buffer_radius` may shrink the geometry completely,
+      in which case an empty geometry is returned. For input the geometries of points or lines,
+      a negative `buffer_radius` will always return an emtpy geometry.
+    impls:
+      - args:
+          - value: geometry
+            name: geom
+          - value: fp64
+            name: buffer_radius
+        return: geometry
+  -
+    name: "centroid"
+    description: >
+      Return a point which is the geometric center of mass of the input geometry.
+    impls:
+      - args:
+          - value: geometry
+            name: geom
+        return: crs_point
```

Comment:
Based on other feedback it seems we should always take in geometry and return geometry types.  Then we don't need to care about the internal behavior of the type at all.

Reply:
sounds good! updated!